src/block/hashtable.rs

```diff
@@ -198,15 +198,26 @@ impl HashTable for HashTableU16 {
 pub fn get_table_size(input_len: usize) -> (usize, usize) {
     let (dict_size, dict_bitshift) = match input_len {
         // U16 Positions
-        0..=500 => (128, 9),
-        501..=1_000 => (256, 8),
-        1_001..=4_000 => (512, 7),
-        4_001..=8_000 => (1024, 6),
-        8_001..=16_000 => (2048, 5),
-        16_001..=65535 => (8192, 3),
+        0..=65535 => {
+            // Considering we want a table with up to 16K bytes and each slot takes 2 bytes.
```

Comment:
Do you mean with each slot, a slot in the hash table?
Because 2 bytes is the case for inputs < U16::MAX, otherwise a slot takes 4bytes.

Therefore, in terms of byte size there is a peak for input sizes between 16384 and 32768.

```
16384 -> 8192, 3
32768 -> 8192, 3
65536 -> 4096, 4
131072 -> 4096, 4
```